cli/src/semgrep/rule_match.py

```diff
@@ -290,6 +284,68 @@ def get_match_based_id(self) -> str:
         match_id_str = str(match_id)
         return f"{hashlib.blake2b(str.encode(match_id_str)).hexdigest()}_{str(self.match_based_index)}"
 
+    @code_hash.default
+    def get_code_hash(self) -> str:
+        """
+        A 32-character hash representation of syntactic_context.
+
+        We started collecting this in addition to syntactic_id because the syntactic_id changes
+        whenever the file path or rule name or index changes, but all of those can be sent in
+        plaintext, so it does not make sense to include them inside a hash.
+
+        By sending the hash of ONLY the code contents, we can determine whether a finding
+        has moved files (path changed), or moved down a file (index changed) and handle that
+        logic in the app. This also gives us more flexibility for changing the definition of
+        a unique finding in the future, since we can analyze things like code, file, index all
+        independently of one another.
+        """
+        hash_int = hash128(str(self.syntactic_context))
```

Comment:
Is murmurhash simply chosen because of historical reasons? If so I'd suggest we switch to something a bit more current and off the shelf like sha256

The murmurhash code looks unmaintained (last commit to the original C library was 8 years ago and we use a pure python re-implementation) and it calls out that the hash is not stable across architectures (32bit vs 64bit produce different values)

Reply:
Good call, thanks for looking into this! Yeah, I just used it because that was what syntactic_id used, but you're right we have an opportunity here to update to something more current, I'll do that.